File: health_system_backend/src/main/java/com/angerasilas/health_system_backend/dto/ProgramInformation.java
package com.angerasilas.health_system_backend.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProgramInformation {
    private Long id;
    private String name;
    private String description;
    private String createdAt;
    private Long createdById;
    private String createdByName;
    private String createdByEmail;
    private List<String> clients;
}

File: health_system_frontend/lib/models/program_in formation.dart
// ignore_for_file: file_names

class ProgramInformation {
  final int id;
  final String name;
  final String description;
  final String createdAt;
  final int createdById;
  final String createdByName;
  final String createdByEmail;
  final List<String> clients;

  ProgramInformation({
    required this.id,
    required this.name,
    required this.description,
    required this.createdAt,
    required this.createdById,
    required this.createdByName,
    required this.createdByEmail,
    required this.clients,
  });

  factory ProgramInformation.fromJson(Map<String, dynamic> json) {
    return ProgramInformation(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      createdAt: json['createdAt'],
      createdById: json['createdById'],
      createdByName: json['createdByName'],
      createdByEmail: json['createdByEmail'],
      clients: List<String>.from(json['clients']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'createdAt': createdAt,
      'createdById': createdById,
      'createdByName': createdByName,
      'createdByEmail': createdByEmail,
      'clients': clients,
    };
  }
}